.travis.yml

```diff
@@ -54,14 +45,11 @@ jobs:
         - LOCALSTACK_HOST=127.0.0.1
         - LOCAL_ES_HOST=127.0.0.1
       install:
-        - ./travis-ci/fetch-cache.sh
         - npm install
         - ln -s /dev/stdout ./lerna-debug.log
         - npm run bootstrap-no-build
         - ./travis-ci/start-local-services.sh
-        - echo 'BOOTSTRAP FINISHED'
-      script: ./node_modules/.bin/nyc ./node_modules/.bin/lerna run test --ignore @cumulus/api
-      after_failure: ./travis-ci/travis_wait_new 50 ./travis-ci/cleanup-integration-tests.sh
```

Comment:
My understanding is that since the stack deployment happens as part of the "unit test and stack deployment" and any step can fail the whole stage, we always want to run the cleanup script on any steps in that stage to be sure that things get cleaned up?

Reply:
Any of the steps failing will cause the entire stage to fail, but the "Deploy Integration Test Stack" is the only one that locks the stack, and my understanding is that the `after_failure` step is like a "finally" block, which will always run if that step has executed.  If I'm missing something, can you lay out a sequence of how things would have to happen where the stack would get locked by the deployment step, but not get unlocked?

Reply:
I see. So you left the `after_failure` step only for "Deploy Integration Test Stack". It seems like that should work.

As an aside, `after_failure` only runs after a failure, but not a success, so it's not quite like `finally`, but still does what we want I think. This documentation was helpful:

https://docs.travis-ci.com/user/job-lifecycle/#the-job-lifecycle

Reply:
> As an aside, `after_failure` only runs after a failure, but not a success, so it's not quite like `finally`, but still does what we want I think.

That's what we want because, if everything succeeds, we want the stack to stay locked while we move on to running the integration tests.

Reply:
Marc, `after_failure` is not like a finally, since it only happens when that step fails.

What would happen is, deployment would start, lock the stack, but eslint would fail and your stack will be locked. I think there could still be some tricky timing issues here, but I think we've chosen to accept edge case scenarios since doing the deployment in parallel brings us a lot of benefit.

Reply:
> What would happen is, deployment would start, lock the stack, but eslint would fail and your stack will be locked.

That makes sense, I'll add the `after_failure` back for the other steps in this stage.

Reply:
Fixed in c00220286